File: sample/dist2/java/src/main/java/org/yanex/hello/IJSII417PublicBaseOfBase.java
package org.yanex.hello;

@javax.annotation.Generated(value = "jsii-pacmak/0.19.0 (build 5d7824d)", date = "2019-10-28T18:25:08.410Z")
public interface IJSII417PublicBaseOfBase extends software.amazon.jsii.JsiiSerializable {

    java.lang.Boolean getHasRoot();

    void foo();

    /**
     * A proxy class which represents a concrete javascript instance of this type.
     */
    final static class Jsii$Proxy extends software.amazon.jsii.JsiiObject implements org.yanex.hello.IJSII417PublicBaseOfBase {
        protected Jsii$Proxy(final software.amazon.jsii.JsiiObjectRef objRef) {
            super(software.amazon.jsii.JsiiObject.InitializationMode.JSII);
            this.setObjRef(objRef);
        }

        @Override
        public java.lang.Boolean getHasRoot() {
            return this.jsiiGet("hasRoot", java.lang.Boolean.class);
        }

        @Override
        public void foo() {
            this.jsiiCall("foo", Void.class);
        }
    }
}

File: sample/dist2/kotlin/src/main/kotlin/hello/IJSII417PublicBaseOfBase.kt
package org.yanex.hello

@javax.annotation.Generated(value = "jsii-pacmak/0.19.0 (build 5d7824d)", date = "2019-10-29T17:33:08.484Z")
interface IJSII417PublicBaseOfBase: software.amazon.jsii.JsiiSerializable {
    val hasRoot: kotlin.Boolean

    fun foo() {
        this.jsiiCall("foo", kotlin.Any::class.java)
    }

    internal class `Jsii$Proxy`: software.amazon.jsii.JsiiObject, org.yanex.hello.IJSII417PublicBaseOfBase {
        override val hasRoot: kotlin.Boolean

        protected constructor(objRef: software.amazon.jsii.JsiiObjectRef) : super(software.amazon.jsii.JsiiObject.InitializationMode.JSII) {
            this.objRef = objRef
            this.hasRoot = this.jsiiCall("hasRoot", kotlin.Boolean::class.java)
        }

        fun foo() {
            this.jsiiCall("foo", kotlin.Any::class.java)
        }
    }
}
